### PR TITLE
[dynamo][collective_op] Use the value of the wrappered attribute async_op in dynamo when checking supported or not

### PR DESCRIPTION
**Pull request by zejun-chen**
I found whatever the attribute `async_op` in collective ops is `True` or `False` explicitly set by the users, it always leads to the graph break because the argument `async_op` is wrappered as `ConstantVariable(bool)` in dynamo. So here we need to use the `value` for the judgement.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Comment by zejun-chen** [comment]
Hi, @wconstab 
Could you help review the PR? Thank you:)

File: torch/_dynamo/variables/functions.py [hunk 1]
```diff
@@ -596,7 +596,7 @@ def call_function(
         # call_function must check any unsupported arguments and graph-break.
         # It's safe to assume args/kwargs from orig_fn map 1:1 to args/kwargs of remapped_fn,
         # since that's the contract for putting a mapping in `traceable_collective_remaps`
-        if kwargs.get("async_op", False):
+        if "async_op" in kwargs and kwargs["async_op"].value:
```
Comment (jansel) [inline torch/_dynamo/variables/functions.py:599]:
```suggestion
        if "async_op" in kwargs and kwargs["async_op"].as_python_constant():
```

**Review by jansel — CHANGES_REQUESTED** [review]
Can you add a test?

**Comment by zejun-chen** [comment]
> Can you add a test?

OK. Sure.

**Review by wconstab — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**